hedera-node/src/main/java/com/hedera/services/context/properties/SemanticVersions.java

```diff
@@ -30,51 +29,72 @@
+					"(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$");
+
+	private static final String HAPI_VERSION_KEY = "hapi.proto.version";
+	private static final String HEDERA_VERSION_KEY = "hedera.services.version";
+
+	private String versionInfoResource = "semantic-version.properties";
 
-	static AtomicReference<ActiveVersions> knownActive = new AtomicReference<>(null);
+	private AtomicReference<ActiveVersions> knownActive = new AtomicReference<>(null);
 
 	public Optional<ActiveVersions> getDeployed() {
 		return Optional.ofNullable(knownActive.get())
 				.or(() -> Optional.ofNullable(fromResource(
-						VERSION_INFO_RESOURCE,
+						versionInfoResource,
 						HAPI_VERSION_KEY,
 						HEDERA_VERSION_KEY)));
 	}
 
-	private static ActiveVersions fromResource(String propertiesFile, String protoKey, String servicesKey) {
+	ActiveVersions fromResource(String propertiesFile, String protoKey, String servicesKey) {
 		try (InputStream in = GetVersionInfoAnswer.class.getClassLoader().getResourceAsStream(propertiesFile)) {
 			var props = new Properties();
 			props.load(in);
 			log.info("Discovered semantic versions {} from resource '{}'", props, propertiesFile);
-			knownActive.set(new ActiveVersions(
-					asSemVer((String)props.get(protoKey)),
-					asSemVer((String)props.get(servicesKey))));
+			var protoSemVer = asSemVer((String) props.get(protoKey));
+			var hederaSemVer = asSemVer((String) props.get(servicesKey));
+			knownActive.set(new ActiveVersions(protoSemVer, hederaSemVer));
 		} catch (Exception surprising) {
 			log.warn(
-					"Failed to read versions from resource '{}' (keys '{}' and '{}')",
+					"Failed to parse resource '{}' (keys '{}' and '{}'). Version info will be unavailable!",
 					propertiesFile,
 					protoKey,
 					servicesKey,
 					surprising);
+			var emptySemver = SemanticVersion.getDefaultInstance();
+			knownActive.set(new ActiveVersions(emptySemver, emptySemver));
 		}
 		return knownActive.get();
 	}
 
-	public static SemanticVersion asSemVer(String value) {
-		long[] parts = asDotDelimitedLongArray(value);
-		return SemanticVersion.newBuilder()
-				.setMajor((int)parts[0])
-				.setMinor((int)parts[1])
-				.setPatch((int)parts[2])
-				.build();
+	SemanticVersion asSemVer(String value) {
+		final var matcher = SEMVER_SPEC_REGEX.matcher(value);
+		if (matcher.matches()) {
+			final var builder = SemanticVersion.newBuilder()
+					.setMajor(Integer.parseInt(matcher.group(1)))
+					.setMinor(Integer.parseInt(matcher.group(2)))
+					.setPatch(Integer.parseInt(matcher.group(3)));
+			if (matcher.group(4) != null) {
+				builder.setPre(matcher.group(4));
+			}
+			if (matcher.group(5) != null) {
+				builder.setBuild(matcher.group(5));
+			}
+			return builder.build();
+		} else {
+			throw new IllegalArgumentException("Argument value='" + value + "' is not a valid semver");
+		}
+	}
```

Comment:
It would be helpful to know different examples of the input string in comment. From the code it seems that part 4 and part 5 are optional so a description here would help.